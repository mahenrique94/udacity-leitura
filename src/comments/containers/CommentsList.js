import PropTypes from 'prop-types'
import React, { Component } from 'react'

import { connect } from 'react-redux'

import { requestEdit, requestGetAll, requestRemove, requestVote } from '../actions'

import { getAllComments, getLoadingComments } from '../selectors'

import CommentsListPage from '../pages/CommentsList'

class CommentsList extends Component {

    static propTypes = {
        edit: PropTypes.func.isRequired,
        getAll: PropTypes.func.isRequired,
        list: PropTypes.arrayOf(PropTypes.object).isRequired,
        loading: PropTypes.bool.isRequired,
        postId: PropTypes.string.isRequired,
        remove: PropTypes.func.isRequired,
        vote: PropTypes.func.isRequired
    }

    componentDidMount() {
        const { getAll, postId } = this.props
        getAll(postId)
    }

    render() {
        const { edit, list, loading, remove, vote } = this.props
        return <CommentsListPage edit={ edit } list={ list } loading={ loading } remove={ remove } vote={ vote }/>
    }

}

const mapStateToProps = state => ({
    list: getAllComments(state),
    loading: getLoadingComments(state)
})

const mapDispatchToProps = {
    edit: id => requestEdit(id),
    getAll: requestGetAll,
    remove: id => requestRemove(id),
    vote: (id, type) => requestVote({ id, type })
}

export default connect(mapStateToProps, mapDispatchToProps)(CommentsList)
