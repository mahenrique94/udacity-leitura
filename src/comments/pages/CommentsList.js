import PropTypes from 'prop-types'
import React, { Component, Fragment } from 'react'

import { not } from 'utils/functions'

import If from 'components/If'
import Loading from 'components/Loading'
import Comment from '../components/comment/Comment'

class CommentsList extends Component {

    static propTypes = {
        edit: PropTypes.func.isRequired,
        list: PropTypes.arrayOf(PropTypes.object).isRequired,
        loading: PropTypes.bool.isRequired,
        remove: PropTypes.func.isRequired,
        vote: PropTypes.func.isRequired
    }

    render() {
        const { list, loading, edit, remove, vote } = this.props
        return (
            <Fragment>
                <If condition={ not(loading) } el={ Loading }>
                    {
                        list.map((comment, index) =>
                            <Comment
                                comment={ comment }
                                data-cy={ `comment-${comment.id}` }
                                edit={ edit }
                                key={ index }
                                remove={ remove }
                                vote={ vote }
                            />
                        )
                    }
                </If>
            </Fragment>
        )
    }

}

export default CommentsList
