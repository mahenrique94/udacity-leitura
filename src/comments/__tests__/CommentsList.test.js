import React from 'react'
import renderer from 'react-test-renderer'

import CommentsList from '../pages/CommentsList'

import CommentModel from '../Comment'

describe('Tests for CommentsList page', () => {

    const REQUIRED_PROPS = {
        edit: () => console.log('Editing...'),
        list: [
            new CommentModel({
                author: 'Test',
                body: 'Testing...',
                deleted: false,
                id: 'abc123',
                parentDeleted: false,
                parentId: '123abc',
                timestamp: 123123123123,
                voteScore: 1
            })
        ],
        loading: false,
        remove: () => console.log('Removing...'),
        vote: () => console.log('Voting...')
    }

    it('Should rendering without crash', () => {
        const tree = renderer.create(
            <CommentsList { ...REQUIRED_PROPS }/>
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('Should rendering a loading component when loading is true', () => {
        const tree = renderer.create(
            <CommentsList { ...REQUIRED_PROPS } loading={ true }/>
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('Should rendering a empty list when list is empty', () => {
        const tree = renderer.create(
            <CommentsList { ...REQUIRED_PROPS } list={ [] }/>
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })

})
