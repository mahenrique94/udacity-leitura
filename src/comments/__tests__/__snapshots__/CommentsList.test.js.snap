// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for CommentsList page Should rendering a empty list when list is empty 1`] = `null`;

exports[`Tests for CommentsList page Should rendering a loading component when loading is true 1`] = `
<div
  className="sc-bdVaJa jUkFeX"
/>
`;

exports[`Tests for CommentsList page Should rendering without crash 1`] = `
<div
  className="mt-5 sc-iwsKbI kRjAdc media"
>
  <div
    className="media-body"
  >
    <h4
      className="media-heading"
    >
      Test
    </h4>
    Testing...
  </div>
  <div
    className="sc-dnqmqq hMmkfQ sc-htoDjs ldLlmj"
  >
    <div
      className="sc-gzVnrw gQVUgH"
    >
      <button
        className="mr-2 btn btn-success btn-sm"
        onClick={[Function]}
        type="button"
      >
        <span
          className="mr-2"
          style={
            Object {
              "verticalAlign": "middle",
            }
          }
        >
          1
        </span>
      </button>
      <button
        className="mr-2 btn btn-danger btn-sm"
        onClick={[Function]}
        type="button"
      />
    </div>
    <button
      className="ml-2 btn btn-warning btn-sm"
      onClick={[Function]}
      type="button"
    />
    <button
      className="ml-2 btn btn-danger btn-sm"
      onClick={[Function]}
      type="button"
    />
  </div>
</div>
`;
